fields.go

```diff
@@ -107,27 +107,21 @@ func extractFields(prefix []string, target interface{}) ([]Field, error) {
 		}
 
 		switch {
+		// If we found a struct that can't deserialize itself, drill down, appending fields as we go.
+		case f.Kind() == reflect.Struct && setterFrom(f) == nil && textUnmarshaler(f) == nil && binaryUnmarshaler(f) == nil:
+			// Prefix for any subkeys is the fieldKey, unless it's
+			// anonymous, then it's just the prefix so far.
+			innerPrefix := fieldKey
+			if structField.Anonymous {
+				innerPrefix = prefix
+			}
 
-		// If we've found a struct, drill down, appending fields as we go.
-		case f.Kind() == reflect.Struct:
```

Comment:
Why does any code need to be deleted? I rather you just add the code you need?

Reply:
I hate this system of PR. What is actually chaning here? Just adding to the case?

Reply:
Lookng at the original code, I don't see any need for this change. The switch is easy to read for match a struct type and the condition make the rest clear.